src/Graphics/Image/Processing/Hough.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | Hough Transform is used as a part of feature extraction in images. 
+-- It is a tool that makes it far easier to identify straight lines in
+-- the source image, whatever their orientation.
+module Graphics.Image.Processing.Hough where
+
+import Control.Monad (forM_, when)
+import qualified Data.Foldable as F (maximum)
+import Data.Array
+import Data.Array.ST (newArray, writeArray, readArray, runSTArray)
+
+import Prelude as P hiding (subtract)
+import Graphics.Image
+import Graphics.Image.Interface as I
+import Graphics.Image.Types as IP
+
+-- | Some helper functions :
+-- | Trivial function for subtracting co-ordinate pairs 
+sub :: Num x => (x, x) -> (x, x) -> (x, x)
+sub (x1, y1) (x2, y2) = (x1 - x2, y1 - y2)
+
+-- | Compute the sum of squares or dot product of a given pair of co-ordinates
+dotProduct :: Num x => (x, x) -> (x, x) -> x
+dotProduct (x1, y1) (x2, y2) = (x1 * x2) + (y1 * y2)
+
+-- | Conversion of pair fromIntegral
+fromIntegralP :: (Integral x, Num y) => (x, x) -> (y, y)
+fromIntegralP (x1, y1) = (fromIntegral x1, fromIntegral y1)
+
+-- | Compute magnitude
+mag :: Floating x => (x, x) -> x
+mag x = sqrt (dotProduct x x)
+
+-- | 'hough' computes the Linear Hough Transform and maps each point in the target image, ​ (ρ, θ) ​ 
+-- to the average color of the pixels on  the corresponding line of the source image ​(x,y) ​- space,
+-- where the line corresponds to points of the form ​(xcosθ + ysinθ = ρ(rho)). 
+--
+-- The idea is that where there is a straight line in the original image, it corresponds to a 
+-- bright (or dark, depending on the color of the background field) spot; by applying a suitable 
+-- filter to the results of the transform, it is possible to extract the locations of the lines in the original image.
+--
+-- <<images/yield.jpg>>
+--
```

Comment:
Could you please include the result image of applying the hough transform to `yield.jpg` here.
This way a user will know what sort of result to expect.

Reply:
Sure sir, will do.

Reply:
Sure sir, will do so.